sdk/communication/Azure.Communication.JobRouter/README.md

```diff
@@ -0,0 +1,298 @@
+        {
+            new WorkerSelector("Some-Skill", LabelOperator.GreaterThan, new LabelValue(10))
+        }
+    });
+```
+
+### Worker
+Now, we register a worker to receive work from that queue, with a label of `Some-Skill` equal to 11.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_RegisterWorker_Async
+var worker = await routerClient.CreateWorkerAsync(
+    new CreateWorkerOptions(
+        workerId: "worker-1",
+        totalCapacity: 1)
+    {
+        QueueIds = new Dictionary<string, QueueAssignment>() { [queue.Value.Id] = new QueueAssignment() },
+        Labels = new Dictionary<string, LabelValue>()
+        {
+            ["Some-Skill"] = new LabelValue(11)
+        },
+        ChannelConfigurations = new Dictionary<string, ChannelConfiguration>()
+        {
+            ["my-channel"] = new ChannelConfiguration(1)
+        },
+        AvailableForOffers = true,
+    }
+);
+```
+
+### Offer
+We should get a [RouterWorkerOfferIssued][offer_issued_event_schema] from our [EventGrid subscription][subscribe_events].
+
+There are several different Azure services that act as a [event handler][event_grid_event_handlers].
+For this scenario, we are going to assume Webhooks for event delivery. [Learn more about Webhook event delivery][webhook_event_grid_event_delivery]
+
+Once events are delivered to the event handler, we can deserialize the JSON payload into a list of events.
+
+```C#
+// Parse the JSON payload into a list of events
+EventGridEvent[] egEvents = EventGridEvent.ParseMany(BinaryData.FromStream(httpContent));
+```
+
+```C#
+string offerId = null;
+foreach (EventGridEvent egEvent in egEvents)
+{
+    // If the event is a system event, TryGetSystemEventData will return the deserialized system event
+    if (egEvent.TryGetSystemEventData(out object systemEvent))
+    {
+        switch (systemEvent)
+        {
+            case SubscriptionValidationEventData subscriptionValidated:
+                Console.WriteLine(subscriptionValidated.ValidationCode);
+                break;
+            ...
+            ...
+            case AcsRouterWorkerOfferIssuedEventData issuedOffer:
+                Console.WriteLine($"Received offer with id: {issuedOffer.OfferId}");
+                offerId = issuedOffer.OfferId;
+                break;
+             ...
+             ...
+             ...
+            // Handle any other system event type
+            default:
+                Console.WriteLine(egEvent.EventType);
+                // we can get the raw Json for the event using Data
+                Console.WriteLine(egEvent.Data.ToString());
+                break;
+        }
+    }
+}
+```
+
+However, we could also wait a few seconds and then query the worker directly against the JobRouter API to see if an offer was issued to it.
+```C# Snippet:Azure_Communication_JobRouter_Tests_Samples_QueryWorker_Async
+var result = await routerClient.GetWorkerAsync(worker.Value.Id);
```

Comment:
We generally encourage avoiding `var` for sample snippets unless it's a direct `new` assignment, since there's no way to inspect the code and understand the type.

Reply:
Leaving the one comment here, but this pattern exists in many of the other samples as well.

Reply:
This has been updated to not use `var` in all samples